src/components/ToggleSwitch.js

```diff
@@ -0,0 +1,172 @@
+      );
+    }
+  }
+
+  render () {
+    const styles = {
+      component: {
+        position: 'relative',
+        fontFamily: StyleConstants.FontFamily,
+        fontSize: '12px',
+        height: this.props.height + 'px',
+        display: 'inline-block'
+      },
+      toggle: {
+        backgroundColor: StyleConstants.Colors.LIGHT_FONT,
+        position: 'relative',
+        borderRadius: this.props.height + 'px',
+        display: 'inline-block',
+        margin: '0 10px',
+        width: this.props.width + 'px',
+        height: this.props.height + 'px',
+        verticalAlign: 'middle',
+        cursor: 'pointer'
+      },
+      left: {
+        top: this.props.sliderInset,
+        left: this.props.sliderInset,
+        transition: 'all .1s'
+      },
+      right: {
+        top: this.props.sliderInset,
+        left: this.props.width - this.props.sliderSize - this.props.sliderInset,
+        transition: 'all .1s'
+      },
+      active: {
+        position: 'absolute',
+        background: StyleConstants.Colors.INVERSE_PRIMARY,
+        borderRadius: '100%',
+        height: (this.props.sliderSize) + 'px',
+        width: (this.props.sliderSize) + 'px'
+      },
+      activeLabel: {
+        color: this.props.activeColor
+      },
+      inactiveLabel: {
+        color: this.props.inactiveColor
+      },
+      label: {
+        cursor: 'pointer',
+        fontWeight: 'bold'
+      }
+    };
+
+    return (
+      <div className='toggle-switch-component' style={[styles.component, this.props.componentStyle]}>
+        {this._renderLeftLabel(styles)}
+        <div className='toggle-switch' onClick={this._handleToggle.bind(this)} style={[styles.toggle, this.props.toggleStyle]} >
+          <div className='toggle-switch-slider' style={[styles.active, this.state.activePosition === 'left' && styles.left || styles.right, this.props.sliderStyle]}></div>
+          {this.props.children}
+        </div>
+        {this._renderRightLabel(styles)}
+      </div>
+    );
+  }
+}
+
+ToggleSwitch.propTypes = {
+  activeColor: React.PropTypes.string,
+  children: React.PropTypes.node,
+  componentStyle: React.PropTypes.oneOfType([
+    React.PropTypes.array,
+    React.PropTypes.object
+  ]),
+  defaultPosition: React.PropTypes.oneOf(['left', 'right']),
+  height: React.PropTypes.number,
+  inactiveColor: React.PropTypes.string,
+  labelStyle: React.PropTypes.oneOfType([
+    React.PropTypes.array,
+    React.PropTypes.object
+  ]),
+  leftLabel: React.PropTypes.string,
+  onToggle: React.PropTypes.func,
+  rightLabel: React.PropTypes.string,
+  showLabels: React.PropTypes.bool,
+  sliderInset: React.PropTypes.number,
+  sliderSize: React.PropTypes.number,
+  sliderStyle: React.PropTypes.oneOfType([
```

Comment:
Just for clarification, should the gray area or the circle that toggles back and forth be the track? I wondered about the naming in this.